### DISCUSSION TITLE
[提问][2.3.6. 变换一个矩阵乘法程序]为什么这里会cache友好？

### DISCUSSION THREAD

**Discussion by hbsun2113** [mlc-ai/mlc-zh#16] [Q&A] [answered]
请问为什么这样的[变换](https://mlc.ai/zh/chapter_tensor_program/index.html#:~:text=yo%2C%20yi%20%3D%20sch.split(y%2C%20%5BNone%2C%20block_size%5D)%0Axo%2C%20xi%20%3D%20sch.split(x%2C%20%5BNone%2C%20block_size%5D)%0A%0Asch.reorder(yo%2C%20xo%2C%20k%2C%20yi%2C%20xi)%0Aprint(sch.mod.script()))会使得cache friendly?
![image](https://user-images.githubusercontent.com/27994442/175809174-426c01ee-7932-4fb7-b182-4c036a1dee83.png)

**Comment by Hzfengsy** [comment, marked as answer]
这里有相关介绍https://tvm.apache.org/docs/how_to/optimize_operators/opt_gemm.html#blocking

**Comment by GeLee-Q** [comment]
因为如果运算被分块的大小能够在一级缓存里进行的话，速度会提升非常快
